pygmt/base_plotting.py

```diff
@@ -1581,405 +1580,5 @@ def text(
-                spec_array = []
-                for index in range(len(spec)):
-                    # Construct the array one row at a time (note that order
-                    # matters here, hence the list comprehension!)
-                    row = [longitude[index], latitude[index], depth[index]] + [
-                        spec[key][index] for key in foc_params
-                    ]
-
-                    # Add in plotting options, if given, otherwise add 0s as
-                    # required by GMT
-                    for arg in plot_longitude, plot_latitude:
-                        if arg is None:
-                            row.append(0)
-                        else:
-                            if "C" not in kwargs:
-                                kwargs["C"] = True
-                            row.append(arg[index])
-                    spec_array.append(row)
-                spec = spec_array
-
-            else:
-                raise GMTError(
-                    "Parameter 'spec' contains values of an unsupported type."
-                )
-
-        # Add condition and scale to kwargs
-        if convention == "aki":
-            data_format = "a"
-        elif convention == "gcmt":
-            data_format = "c"
-        elif convention == "mt":
-            # Check which component of mechanism the user wants plotted
-            if component == "deviatoric":
-                data_format = "z"
-            elif component == "dc":
-                data_format = "d"
-            else:  # component == 'full'
-                data_format = "m"
-        elif convention == "partial":
-            data_format = "p"
-        elif convention == "principal_axis":
-            # Check which component of mechanism the user wants plotted
-            if component == "deviatoric":
-                data_format = "t"
-            elif component == "dc":
-                data_format = "y"
-            else:  # component == 'full'
-                data_format = "x"
-        # Support old-school GMT format options
-        elif convention in ["a", "c", "m", "d", "z", "p", "x", "y", "t"]:
-            data_format = convention
-        else:
-            raise GMTError("Convention not recognized.")
-
-        # Assemble -S flag
-        kwargs["S"] = data_format + scale
-
-        kind = data_kind(spec)
-        with Session() as lib:
-            if kind == "matrix":
-                file_context = lib.virtualfile_from_matrix(np.atleast_2d(spec))
-            elif kind == "file":
-                file_context = dummy_context(spec)
-            else:
-                raise GMTInvalidInput("Unrecognized data type: {}".format(type(spec)))
-            with file_context as fname:
-                arg_str = " ".join([fname, build_arg_string(kwargs)])
-                lib.call_module("meca", arg_str)
+    # GMT Supplementary modules
+    from pygmt.src.meca import meca  # pylint: disable=import-outside-toplevel
```

Comment:
> is this as "simple" as moving a single function to it's own script and adding "from pygmt.src._script_ import _script function_? As far as I can tell, that is the case, but I want to check with you before doing work/submitting pull requests for the other functions, especially those that I'm in the process of wrapping.

Yes, this is the key line. It looks simple and works, but:

1) we would need to write a separate line for each new module (e.g. `velo`, `solar`, etc).
2) I don't really like having the pylint `disable=import-outside-toplevel` here.

To be honest, there's probably a better way to do this. Any good ideas? Or we can just settle for this for now.

Reply:
Is it possible to have `from pygmt.src.meca import meca` in `pygmt/src/__init__.py`, then we only need `from pygmt.src import meca, velo, text` here.

Reply:
Yep sure, I'll do that, but it would require changing two files every time we wrap a new module (instead of one right now).